src/main/java/org/reaktivity/nukleus/http2/internal/types/stream/Http2PingFW.java

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright 2016-2017 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.nukleus.http2.internal.types.stream;
+
+import org.agrona.DirectBuffer;
+import org.agrona.MutableDirectBuffer;
+import org.reaktivity.nukleus.http2.internal.types.Flyweight;
+
+import java.nio.ByteOrder;
+
+import static java.nio.ByteOrder.BIG_ENDIAN;
+import static org.reaktivity.nukleus.http2.internal.types.stream.Http2Flags.ACK;
+import static org.reaktivity.nukleus.http2.internal.types.stream.Http2FrameType.PING;
+
+/*
+
+    Flyweight for HTTP2 PING frame
+
+    +-----------------------------------------------+
+    |                 Length (24)                   |
+    +---------------+---------------+---------------+
+    |   Type (8)    |   Flags (8)   |
+    +-+-------------+---------------+-------------------------------+
+    |R|                 Stream Identifier (31)                      |
+    +=+=============+===============================================+
+    |                                                               |
+    |                      Opaque Data (64)                         |
+    |                                                               |
+    +---------------------------------------------------------------+
+
+ */
+public class Http2PingFW extends Flyweight
+{
+    private static final int LENGTH_OFFSET = 0;
+    private static final int TYPE_OFFSET = 3;
+    private static final int FLAGS_OFFSET = 4;
+    private static final int STREAM_ID_OFFSET = 5;
+    private static final int PAYLOAD_OFFSET = 9;
+
+    public int payloadOffset()
+    {
+        return offset() + PAYLOAD_OFFSET;
+    }
+
+    public int payloadLength()
+    {
+        int length = (buffer().getByte(offset() + LENGTH_OFFSET) & 0xFF) << 16;
+        length += (buffer().getByte(offset() + LENGTH_OFFSET + 1) & 0xFF) << 8;
+        length += buffer().getByte(offset() + LENGTH_OFFSET + 2) & 0xFF;
+
+        // assert length == 8; caller needs to validate
```

Comment:
I will remove assert as it is checked by the caller. There are two options:
1. Throw exception and caller catches and sends appropriate error on the connection
2. Or caller verifies the necessary contracts and sends appropriate error on the connection. In some cases, this becomes a problem when the flyweights are deeply nested.

What is the preferred way in flyweights ?